### PR TITLE
Rel14

### PR DESCRIPTION
**Pull request by em-eight**
Add REL_14 relocation support and print missing symbols as a warning when _ignore_missing is true

### CODE REVIEW

File: ppcdis/elf2rel.py [hunk 1]
```diff
@@ -111,6 +111,8 @@ def _get_sections_to_link(self) -> List[int]:
             if sec["sh_type"] in ("SHT_PROGBITS", "SHT_NOBITS")
             and sec["sh_flags"] & SH_FLAGS.SHF_ALLOC
             and sec.name not in  ("forcestrip", "relsymdef")
+            # TODO: link sections by name instead of index
+            and sec.name in (".text", ".ctors", ".dtors", ".data", ".rodata", ".bss")
```
Comment (SeekyCt) [inline ppcdis/elf2rel.py:115]: What's the point of this?

**Comment by SeekyCt** [comment]
Just out of curiosity, when does rel14 actually get emitted?

File: ppcdis/elf2rel.py (hunk 1 above)
Comment (em-eight) [reply ppcdis/elf2rel.py:115]: Prevents sections other that those from being created. mkw decomp has a helper "binary_blobs" section that looks like a data section so `elf2rel.py` emits it. But yeah it's pretty bad code if the REL format can contain other sections.

Comment (SeekyCt) [reply ppcdis/elf2rel.py:115]: I think a better solution might be having a command line flag where you can specify extra section names to skip, I've not saw a real rel use a custom section yet but they definitely could. I don't mind just adding that myself after merging if you'd like. Not sure what you mean by the todo btw?

**Comment by SeekyCt** [comment]
The only time I ever saw them was with GAS when doing conditional branches to global labels, which shouldn't really happen (this is why I use separate labels for jumptable targets), but there's no harm in supporting it anyway

File: ppcdis/elf2rel.py (hunk 1 above)
Comment (em-eight) [reply ppcdis/elf2rel.py:115]: agreed, I will commit some time today. The todo is from when I didn't know to how to force section order with -r1, I'll delete it.

**Comment by SeekyCt** [comment]
Thanks, I'll take a look in the morning

**Pull request merged**